components/kyma-environment-broker/internal/process/upgrade_kyma/lms_cert.go

```diff
@@ -0,0 +1,233 @@
+	tenantReadyRetryInterval = 30 * time.Second
+	lmsTimeout               = 30 * time.Minute
+	upgradekibanaURLLabelKey = "operator_lmsUrl"
+)
+
+type LmsClient interface {
+	RequestCertificate(tenantID string, subject pkix.Name) (id string, privateKey []byte, err error)
+	GetCertificateByURL(url string) (cert string, found bool, err error)
+	GetCACertificate(tenantID string) (cert string, found bool, err error)
+	GetTenantStatus(tenantID string) (status lms.TenantStatus, err error)
+	GetTenantInfo(tenantID string) (status lms.TenantInfo, err error)
+}
+
+type LMSUpgradeStep struct {
+	operationManager *process.UpgradeKymaOperationManager
+	isMandatory      bool
+	expirationTime   time.Duration
+}
+
+type lmsCertStep struct {
+	LMSUpgradeStep
+	provider            LmsClient
+	normalizationRegexp *regexp.Regexp
+}
+
+func NewLmsCertificatesStep(certProvider LmsClient, os storage.Operations, isMandatory bool) *lmsCertStep {
+	return &lmsCertStep{
+		LMSUpgradeStep: LMSUpgradeStep{
+			operationManager: process.NewUpgradeKymaOperationManager(os),
+			isMandatory:      isMandatory,
+			expirationTime:   lmsTimeout,
+		},
+		provider:            certProvider,
+		normalizationRegexp: regexp.MustCompile("[^a-zA-Z0-9]+"),
+	}
+}
+
+func (s *lmsCertStep) Name() string {
+	return "Upgrade_Request_LMS_Certificates"
+}
+
+// Run executes getting LMS certificates steps, which means:
+// 1. check if the tenant is ready
+// 2. request certificates
+// 3. poll CA and signed certificates
+func (s *lmsCertStep) Run(operation internal.UpgradeKymaOperation, l logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	if operation.Lms.Failed {
+		l.Info("LMS has failed, skipping")
+		return operation, 0, nil
+	}
+	logger := l.WithField("LMSTenant", operation.Lms.TenantID)
+
+	if operation.Lms.TenantID == "" {
+		logger.Error("Create LMS Tenant step must be run before")
+		return operation, 0, errors.New("the step needs to be run after 'Create LMS tenant' step")
+	}
+
+	// check if LMS tenant is ready
+	status, err := s.provider.GetTenantStatus(operation.Lms.TenantID)
+	if err != nil {
+		return s.handleError(
+			operation,
+			logger,
+			time.Since(operation.Lms.RequestedAt),
+			"Unable to get LMS Tenant status",
+			err)
+	}
+	if !(status.ElasticsearchDNSResolves && status.KibanaDNSResolves) {
+		logger.Infof("LMS tenant not ready: elasticDNS=%v, kibanaDNS=%v", status.ElasticsearchDNSResolves, status.KibanaDNSResolves)
+		if time.Since(operation.Lms.RequestedAt) > lmsTimeout {
+			logger.Error("Setting LMS operation failed - tenant provisioning timed out")
+			return s.failLmsAndUpdate(operation, "LMS Tenant provisioning timeout", l)
+		}
+		return operation, tenantReadyRetryInterval, nil
+	}
+
+	tenantInfo, err := s.provider.GetTenantInfo(operation.Lms.TenantID)
+	if err != nil {
+		return s.handleError(
+			operation,
+			logger,
+			time.Since(operation.Lms.RequestedAt),
+			"Unable to get LMS Tenant info",
+			err)
+	}
+
```

Comment:
we can assume the LMS tenant is created so we do not need to check the tenant status. It seems all lines before the comment could be removed.